### ISSUE TITLE
[Question] Keep sincronization with data generated at different rate

### ISSUE THREAD

**Issue by torydebra** [ADVRHumanoids/MatLogger2#8]
Hi,

I have some data **A** that is generated at each loop
Then I have another data **B** that sometimes is not generated

So, if I log data B _only when it is generated_, I loose the _time syncronization_ with data A, because B will have less columns (where the "time" is) and I can not understand which column B generated at loop X correspond to the column of A generated at the same loop X

A solution can be to generated always also B data, putting zeros as value when I do not have values. 

Have you used different approaches? @liesrock @alaurenzi ?

**Comment by liesrock** [comment]
if you log the "real" time and you plot A and B w.r.t time you have A and B synchronized even if they do not have the same dimension.

**Comment by alaurenzi** [comment]
Yes, recording separate timestamps for both series is an option. You can try (I have never done it) to record `NaN`s instead of zeros, so that matlab won't show them in the plot.

**Comment by torydebra** [comment]
So,
- I would need a timestamp for each "not always available" variables plus one timestamp for all the "always available" ones
- Log everything adding NaN or some "invalid" value

Is there the possibility in the future to have timestamps directly integrated in the logged variable?
